src/devices/Ahtxx/AhtBase.cs

```diff
@@ -0,0 +1,180 @@
+                (byte)Command.Measure,
+                0x33,
+                0x00
+            };
+            _i2cDevice.Write(buffer);
+
+            // According to the datasheet the measurement takes 80 ms and completion is indicated by the status bit.
+            // However, it seems to be faster at around 10 ms and sometimes up to 50 ms.
+            while (IsBusy())
+            {
+                Thread.Sleep(10);
+            }
+
+            buffer = stackalloc byte[6];
+            _i2cDevice.Read(buffer);
+
+            // data format: 20 bit humidity, 20 bit temperature
+            // 7               0 7              0 7             4           0 7          0 7         0
+            // [humidity 19..12] [humidity 11..4] [humidity 3..0|temp 19..16] [temp 15..8] [temp 7..0]
+            // c.f. datasheet ch. 5.4.5
+            Int32 rawHumidity = (buffer[1] << 12) | (buffer[2] << 4) | (buffer[3] >> 4);
+            Int32 rawTemperature = ((buffer[3] & 0xF) << 16) | (buffer[4] << 8) | buffer[5];
+            // RH[%] = Hraw / 2^20 * 100%, c.f. datasheet ch. 6.1
+            _humidity = (rawHumidity * 100.0) / 0x100000;
+            // T[°C] = Traw / 2^20 * 200 - 50, c.f. datasheet ch. 6.1
+            _temperature = ((rawTemperature * 200.0) / 0x100000) - 50;
+        }
+
+        /// <summary>
+        /// Perform soft reset command sequence
+        /// </summary>
+        private void SoftReset()
+        {
+            _i2cDevice.WriteByte((byte)Command.SoftRest);
+            // reset requires 20ms at most, c.f. datasheet version 1.1, ch. 5.5
+            Thread.Sleep(20);
+        }
+
+        /// <summary>
+        /// Perform calibration command sequence
+        /// </summary>
+        private void Calibrate()
+        {
+            Span<byte> buffer = stackalloc byte[3]
+            {
+                (byte)Command.Calibrate,
+                0x08, // command parameters c.f. datasheet, version 1.1, ch. 5.4
+                0x00
+            };
+            _i2cDevice.Write(buffer);
+            // wait 10ms c.f. datasheet, version 1.1, ch. 5.4
+            Thread.Sleep(10);
+        }
+
+        private byte GetStatusByte()
+        {
+            _i2cDevice.WriteByte(0x71);
+            // whithout this delay the reading the status fails often.
+            Thread.Sleep(10);
+            byte status = _i2cDevice.ReadByte();
+            return status;
+        }
+
+        private bool IsBusy()
+        {
+            return (GetStatusByte() & (byte)StatusBit.Busy) == (byte)StatusBit.Busy;
+        }
+
+        private bool IsCalibrated()
+        {
+            return (GetStatusByte() & (byte)StatusBit.Calibrated) == (byte)StatusBit.Calibrated;
+        }
+
+        /// <inheritdoc cref="IDisposable" />
+        public void Dispose() => Dispose(true);
+
+        /// <inheritdoc cref="IDisposable" />
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_i2cDevice == null)
```

Comment:
nit: this is not needed since `?.` will handle the null check

Reply:
Correct. Done.